src/queryService.js

```diff
@@ -0,0 +1,149 @@
+const logQuery = async (url, searchParams) => {
+  console.log(searchParams);
+
+  const featureSet = await queryService(url, {
+    searchParams,
+  }).json();
+
+  if (featureSet.error) {
+    throw new Error(featureSet.error.message);
+  }
+
+  return featureSet;
+};
+
+export function lookupSponsor(metadata, layerName, code) {
+  const sponsors = metadata.fields.filter((field) => field.name === layerName)[0].domain.codedValues;
+  const sponsor = sponsors.filter((sponsor) => sponsor.code === code);
+
+  return sponsor.length > 0 ? sponsor[0].name : 'unknown';
+}
+
+export async function getLayerMetadata(layerId) {
+  const metadata = await queryService(layerId.toString()).json();
+
+  return metadata;
+}
+
+export async function getNewProjects() {
+  const featureSet = await logQuery('0/query', {
+    f: 'json',
+    where: `created_date BETWEEN TIMESTAMP '${getBeginningOfLastWeek(new Date())}' AND TIMESTAMP '${getEndOfLastWeek(
+      new Date()
+    )}'`,
+    outFields: 'ProjectID,sponsor,comment_deadline,project_abstract,title_action,county,created_date',
+    orderByFields: 'created_date ASC',
+    returnGeometry: false,
+    resultRecordCount: maxRecordCount,
+  });
+
+  let metadata;
+  if (featureSet.features.length > 0) {
+    metadata = await getLayerMetadata(0);
+  }
+
+  const newProjects = featureSet.features.map((feature) => {
+    return {
+      id: feature.attributes.ProjectID,
+      title: feature.attributes.title_action,
+      abstract: feature.attributes.project_abstract,
+      sponsor: lookupSponsor(metadata, 'sponsor', feature.attributes.sponsor),
+      commentDeadline: `${format(new Date(feature.attributes.comment_deadline), 'MM/dd/yyyy')}`,
+      createdDate: `${format(new Date(feature.attributes.created_date), 'MM/dd/yyyy')}`,
+      county: feature.attributes.county,
+    };
+  });
+
+  return [newProjects, featureSet.features.length];
+}
+
+export async function getUpcomingProjects() {
+  const featureSet = await logQuery('0/query', {
+    f: 'json',
+    where: `comment_deadline>=TIMESTAMP '${getToday(new Date())}'`,
+    outFields: 'ProjectID,sponsor,comment_deadline,project_abstract,title_action',
+    orderByFields: 'comment_deadline ASC',
+    returnGeometry: false,
+    resultRecordCount: maxRecordCount,
+  });
+
+  let metadata;
+  if (featureSet.features.length > 0) {
+    metadata = await getLayerMetadata(0);
+  }
+
+  const projectsByDate = {};
+
+  featureSet.features.forEach((feature) => {
+    const date = format(new Date(feature.attributes.comment_deadline), 'MM/dd/yyyy');
+    if (!projectsByDate[date]) {
+      projectsByDate[date] = [];
+    }
+
+    projectsByDate[date].push({
+      id: feature.attributes.ProjectID,
+      abstract: feature.attributes.project_abstract,
+      title: feature.attributes.title_action,
+      sponsor: lookupSponsor(metadata, 'sponsor', feature.attributes.sponsor),
+      commentDeadline: date,
+    });
+  });
+
+  const keys = Object.keys(projectsByDate).sort();
```

Comment:
**question**: Do you really want these sorted by `MM/dd/yyyy`? Maybe it should be `yyyy/MM/dd`?

Reply:
I think MM/dd/yyyy should be ok since it's only sorting a weeks worth of data. I suppose it could be weird if there were some month spans or around new years. let me think about that...